recipes/management/increasing_the_capacity_to_win/reward_systems/compensation/philosophy.md

```diff
@@ -34,64 +47,83 @@ of product/market fit risk reduction."
-       worth it?
-1. Write up a [Proposal](/tools/written_proposal.md).
-    1. Can you express your pros & cons in an unbiased way?
-1. Get buy-in from stakeholdes (CEO)
-1. Roll out to your team.
-    1. Otherwise, the philosophy isn't real. This is the hardest part!
-
-### Identifying the Problem
-
-Life isn't fair!
-
-### Asking the Right Question
-
-1. What is the maximum the company can afford?
-1. What is the minimum we pay employees? 50% of market? 25%?
-1. What is the maximum? 50%? 75%? 90%?
-1. BANDS:
-    1. Startups: You want to pay the minimum you can get away with. Extends the runway.
-
-1. Do we break the bands? If so, when/how?
-1. How do we want bands to overlap across levels?
-    1. This is important re: promotions.
-1. How do we pay ICs vs managers?
-1. Do we have levels?
-1. Cash/equity balance?
-1. Planned Bonuses?
-1. Unplanned bonuses?
-    1. Yearly?
-    1. Ad-hoc?
-1. What is the value of the equity?
-    1. Founders are thinking of a $10B company... so the equity is HUGE! They have to think this way.
-1. Vesting schedule
-1. Vacation / PTO / etc
-    1. Startups are doing UNLIMITED because that avoids debt on their books (!!)
-1. How often do we change our bands?
-
-# Unorganized
-
--Different job families should have different philosophies. No reason to pay your Ops team crazy equity. Or some of your
-FE roles may not be that important?
-
-Pay everyone top dollar --> cuts the company runway + dillutes equity a ton.
-Pay everyone top equity, low $ --> unable to hire some of the talent you need (only high-risk people), especially senior
-folks (more likely to have families/other obligations).
-
-No philosophy --> might be right in the beginning. "Do whatever it takes" --> management debt.
+Here is the outline of my recipe. I'll add more details as requested by folks.
+
+1. Identify the technical problems that need [A-level talent](/recipes/management/increasing_the_capacity_to_win/hiring/talent.md#A-LevelTalent)
+    1.  I used to believe that you wanted to hire the best possible engineers for every role. That was before I met my first startup engineering budget... which is *slim*. When it turns out those perfect resumes cost 30% more (some maybe even 2-3x more!!), then you start asking: "is this a problem that a B-player can solve?"
+    1. Every startup has 'end-of-the-business' technical problems, and 'good-enough' technical problems. Hopefully, a LOT more of the latter than the former. If you can segregate your problems, and therefore your team quality, you should have a comp philosophy that reflects where it makes sense to invest your $.
+    1. I use job roles for this segregation. For example, we might pay everyone at the same band levels (50-75%, see below), but we divide engineers into 'Application Engineer', 'Fullstack Engineer', 'Backend Engineer', and 'ML Engineer', with each job family being comp'd higher than the previous one.
```

Comment:
Definitely between ML and other roles -- ML engineers tend to make significantly more (10-20%). Frontend engineers tend to make less; "full-stack" is usually a code word for "frontend engineers that dabble in Python." I generally wouldn't trust my hardest infrastructure ('backend') problems to "full-stack engineers" because they will literally just make things worse :) Perhaps I'm biased, as an infra person... but it real infrastructure work requires a level of knowledge that very few 'product engineers' have.

That said, 'backend engineer' is probably the wrong term; I changed it to 'infrastructure engineer'.

Reply:
Makes sense... Still, I wonder if with your transparent compensation philosophy and startup aptitude of everyone should jump to have most impact I'd make it so explicit (that is, you may have separate bands / recs for ML vs non, but keeping 1 SWE ladder with wide enough bands to not introduce too much granularity and controversy).

To our y-day discussion - you may hire an ML eng and have them do infra or product for 6-9 months or hire highly skilled full-stack who actually quickly learns to be infra eng too... 

Regardless, I rest my case here:)